File: FootScout-Vue.WebAPI/Entities/PlayerOffer.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace FootScout_Vue.WebAPI.Entities
{
    public class PlayerOffer
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int ClubAdvertisementId { get; set; }

        [ForeignKey("ClubAdvertisementId")]
        public virtual ClubAdvertisement ClubAdvertisement { get; set; }

        [Required]
        public int OfferStatusId { get; set; }

        [ForeignKey("OfferStatusId")]
        public virtual OfferStatus OfferStatus { get; set; }

        [Required]
        public int PlayerPositionId { get; set; }

        [ForeignKey("PlayerPositionId")]
        public virtual PlayerPosition PlayerPosition { get; set; }

        [Required]
        public int Age { get; set; }

        [Required]
        public int Height { get; set; }

        [Required]
        public int PlayerFootId { get; set; }

        [ForeignKey("PlayerFootId")]
        public virtual PlayerFoot PlayerFoot { get; set; }

        [Required]
        public double Salary { get; set; }

        [MaxLength(200)]
        public string AdditionalInformation { get; set; }

        [Required]
        [DataType(DataType.DateTime)]
        public DateTime CreationDate { get; set; }

        [Required]
        public string PlayerId { get; set; }

        [ForeignKey("PlayerId")]
        public virtual User Player { get; set; }
    }
}

File: FootScout-Vue.WebClient/src/models/interfaces/PlayerOffer.ts
import type { ClubAdvertisement } from './ClubAdvertisement';
import type { OfferStatus } from './OfferStatus';
import type { PlayerPosition } from './PlayerPosition';
import type { PlayerFoot } from './PlayerFoot';
import type { UserDTO } from '../dtos/UserDTO';

export interface PlayerOffer {
    id: number;
    clubAdvertisementId: number;
    clubAdvertisement: ClubAdvertisement;
    offerStatusId: number;
    offerStatus: OfferStatus;
    playerPositionId: number;
    playerPosition: PlayerPosition;
    age: number;
    height: number;
    playerFootId: number;
    playerFoot: PlayerFoot;
    salary: number;
    additionalInformation: string;
    creationDate: string;
    playerId: string;
    player: UserDTO;
}